code/iaas/ha/src/main/java/io/cattle/platform/ha/monitor/impl/PingInstancesMonitorImpl.java

```diff
@@ -344,17 +354,20 @@ protected ReportedInstances getInstances(Ping ping) {
     }
 
     protected AgentAndHost loadAgentAndHostData(ImmutablePair<Long, String> agentIdAndHostUuid) {
-        if (agentIdAndHostUuid.left == null || StringUtils.isEmpty(agentIdAndHostUuid.right))
-            return null;
+        Long agentId = agentIdAndHostUuid.left;
+        String hostUuid = agentIdAndHostUuid.right;
+
+        Agent agent = objectManager.loadResource(Agent.class, agentId);
+        Host host = null;
+        Map<String, Host> hosts = null;
+        if (agent != null) {
+            hosts = agentDao.getHosts(agent.getId());
```

Comment:
We can optimize here and introduce new Dao method searching by agentId/hostUUID. So there would be no need for line 365.